File: server/datasets/java_source-code/Multiple-G3/renamed_41_dsu.java
public class DSU_5 {
    public static void main(String[] args) {
        DSU_5 dsu = new DSU_5(10);
        dsu.u_5(1, 2);
    }
    int[] p_5;
    int[] r_5;
    public DSU_5(int tot_5) {
        p_5 = new int[tot_5];
        r_5 = new int[tot_5];
        for (int idx = 0; idx < tot_5; idx++) {
            p_5[idx] = idx;
            r_5[idx] = 0;
        }
    }
    public int f_5(int x_5) {
        if (p_5[x_5] != x_5) {
            p_5[x_5] = f_5(p_5[x_5]);
        }
        return p_5[x_5];
    }
    public void u_5(int a_5, int b_5) {
        int ra_5 = f_5(a_5);
        int rb_5 = f_5(b_5);
        if (ra_5 != rb_5) {
            if (r_5[ra_5] < r_5[rb_5]) {
                p_5[ra_5] = rb_5;
            } else if (r_5[ra_5] > r_5[rb_5]) {
                p_5[rb_5] = ra_5;
            } else {
                p_5[rb_5] = ra_5;
                r_5[ra_5]++;
            }
        }
    }
}

File: server/datasets/controlled_benchmarks/python/multiple_files/multiple_g3/renamed_41_dsu.py
class DSU_5:
    def __init__(self, n_5):
        self.p_5 = list(range(n_5))
        self.rk_5 = [0] * n_5

    def find_5(self, x_5):
        if self.p_5[x_5] != x_5:
            self.p_5[x_5] = self.find_5(self.p_5[x_5])
        return self.p_5[x_5]

    def union_5(self, x_5_a, x_5_b):
        ra_5 = self.find_5(x_5_a)
        rb_5 = self.find_5(x_5_b)
        if ra_5 != rb_5:
            if self.rk_5[ra_5] < self.rk_5[rb_5]:
                ra_5, rb_5 = rb_5, ra_5
            self.p_5[rb_5] = ra_5
            if self.rk_5[ra_5] == self.rk_5[rb_5]:
                self.rk_5[ra_5] += 1
            return True
        return False
